framework/include/vectorpostprocessors/LineMaterialSamplerBase.h

```diff
@@ -179,6 +184,7 @@ LineMaterialSamplerBase<T>::execute()
       addSample(_q_point[qp], qp_proj_dist_along_line, values);
     }
   }
+  _fe_problem.clearActiveMaterialProperties(_tid);
```

Comment:
Yeah - you only get screwed when you make your own loop. All framework loops are handled automatically. It's really hard to warn unless we take away people's ability to make their own loops. That's probably not going to happen.